utils/rowgroup/rowaggregation.cpp

```diff
@@ -2045,8 +2045,9 @@ void RowAggregation::doStatistics(const Row& rowIn, int64_t colIn, int64_t colOu
         case execplan::CalpontSystemCatalog::UDECIMAL:  // handle scale later
             if (LIKELY(fRowGroupIn.getColumnWidth(colIn) == datatypes::MAXDECIMALWIDTH))
             {
-                int128_t* val128InPtr = rowIn.getBinaryField<int128_t>(colIn);
-                valIn = Dec::getLongDoubleFromWideDecimal(*val128InPtr);
+                // To save from unaligned memory
+                datatypes::TSInt128 val128In(rowIn.getBinaryField<int128_t>(colIn));
+                valIn =  static_cast<long double>(val128In.toTFloat128());
```

Comment:
The `.toTFloat128()` call here is unnecessary.

Reply:
Not exactly. TSInt128::operator long double() shouldn't be used here b/c I wish  to remove all casts from int128 -> real types b/c they doesn't look right. TFloat128 should be used instead. We'll have extra 16 byte asignment step here but we won't be casting from int128 into a real type.